geode-core/src/distributedTest/java/org/apache/geode/internal/cache/ClearGIIDUnitTest.java

```diff
@@ -0,0 +1,507 @@
+      PauseDuringGIICallback myAfterReceivedImageReply =
+          // using bucket name for region name to ensure callback is triggered
+          new PauseDuringGIICallback(
+              InitialImageOperation.GIITestHookType.AfterReceivedRequestImage, "_B__testPR_9");
+      InitialImageOperation.setGIITestHook(myAfterReceivedImageReply);
+    });
+
+    cluster.stop(2);
+
+    createDelta(50);
+
+    restartServerOnVM(2);
+
+    AsyncInvocation async = createRegionAsync(serverVMs[1]);
+    invokeClear(serverVMs[0]);
+
+    serverVMs[1].invoke(() -> InitialImageOperation.resetGIITestHook(
+        InitialImageOperation.GIITestHookType.AfterReceivedRequestImage,
+        true));
+    try {
+      async.getResult(30000);
+    } catch (InterruptedException ex) {
+      Assert.fail("Async create interupted" + ex.getMessage());
+    }
+
+    verifyRegionSizes(0);
+  }
+
+  @Test
+  public void GIICompletesSuccessfullyWhenRunningDuringClearWithThirdDatastore() {
+    startServers(NUM_SERVERS + 1);
+    verifyRegionSizes(DATA_SIZE);
+
+    // set tesk hook
+    serverVMs[1].invoke(() -> {
+      PauseDuringGIICallback myAfterReceivedImageReply =
+          // using bucket name for region name to ensure callback is triggered
+          new PauseDuringGIICallback(
+              InitialImageOperation.GIITestHookType.AfterReceivedRequestImage, "_B__testPR_9");
+      InitialImageOperation.setGIITestHook(myAfterReceivedImageReply);
+    });
+
+    cluster.stop(2);
+
+    createDelta(50);
+
+    restartServerOnVM(2);
+
+    AsyncInvocation async = createRegionAsync(serverVMs[1]);
+    invokeClear(serverVMs[0]);
+
+    serverVMs[1].invoke(() -> InitialImageOperation.resetGIITestHook(
+        InitialImageOperation.GIITestHookType.AfterReceivedRequestImage,
+        true));
+    try {
+      async.getResult(30000);
+    } catch (InterruptedException ex) {
+      Assert.fail("Async create interupted" + ex.getMessage());
+    }
+
+    verifyRegionSizes(0);
+  }
+
+  @Test
+  public void GIICompletesSuccessfullyWhenRunningDuringClearWithPersistentRegion() {
+    if (regionShortcut == RegionShortcut.PARTITION) {
+      regionShortcut = RegionShortcut.PARTITION_PERSISTENT;
+    } else if (regionShortcut == RegionShortcut.REPLICATE) {
+      regionShortcut = RegionShortcut.REPLICATE_PERSISTENT;
+    }
+    startServers(NUM_SERVERS);
+
+    verifyRegionSizes(DATA_SIZE);
+
+    // set tesk hook
+    serverVMs[1].invoke(() -> {
+      PauseDuringGIICallback myAfterReceivedImageReply =
+          // using bucket name for region name to ensure callback is triggered
+          new PauseDuringGIICallback(
+              InitialImageOperation.GIITestHookType.AfterReceivedRequestImage, "_B__testPR_9");
+      InitialImageOperation.setGIITestHook(myAfterReceivedImageReply);
+    });
+
+    cluster.stop(2);
+
+    createDelta(50);
+
+    restartServerOnVM(2);
+
+    AsyncInvocation async = createRegionAsync(serverVMs[1]);
+    invokeClear(serverVMs[0]);
```

Comment:
you need to add waitForCallbackStarted(serverVMs[1], InitialImageOperation.GIITestHookType.AfterReceivedRequestImage); 
before calling clear. This call will make sure the GII stopped at the certain stage. 

I added it for you and it failed. The reason is: you changed receivedImageReply to be AfterReceivedRequestImage. 

Only the provider will receive AfterReceivedRequestImage.

You need to do this for all of your test cases.